conan/tools/cmake/toolchain.py

```diff
@@ -394,28 +395,84 @@ def context(self):
         return ctxt_toolchain
 
 
-class FindConfigFiles(Block):
+class FindFiles(Block):
     template = textwrap.dedent("""
         {% if find_package_prefer_config %}
         set(CMAKE_FIND_PACKAGE_PREFER_CONFIG {{ find_package_prefer_config }})
         {% endif %}
-        # To support the generators based on find_package()
-        {% if cmake_module_path %}
-        set(CMAKE_MODULE_PATH {{ cmake_module_path }} ${CMAKE_MODULE_PATH})
+        {% if generators_folder or host_build_paths_root or host_build_paths_noroot or build_build_paths %}
+        set(CMAKE_MODULE_PATH {{ generators_folder }} {{ host_build_paths_root }} {{ host_build_paths_noroot }} {{ build_build_paths }} ${CMAKE_MODULE_PATH})
         {% endif %}
-        {% if cmake_prefix_path %}
-        set(CMAKE_PREFIX_PATH {{ cmake_prefix_path }} ${CMAKE_PREFIX_PATH})
+        {% if generators_folder or host_build_paths_noroot %}
```

Comment:
Why `host_build_paths_noroot` should be set to `CMAKE_PREFIX_PATH`? just in case a package contains `config` files?  In that case, what if a user wanted to package a `config` file located in the root?

Reply:
> Why host_build_paths_noroot should be set to CMAKE_PREFIX_PATH? just in case a package contains config files?

Yes

>  In that case, what if a user wanted to package a config file located in the root?

It wouldn't be discoverable, and this toolchain can't allow to add root package folder of requires to `CMAKE_PREFIX_PATH`, otherwise executables in `<root>/bin` of requires are discoverable before executables of build_requires (`CMAKE_PREFIX_PATH` is searched before `CMAKE_PROGRAM_PATH`, and `CMAKE_PROGRAM_PATH` before `PATH`).